docker/rockylinux8/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM rockylinux:8.5.20220308
+RUN dnf install -y 'dnf-command(config-manager)'
+RUN dnf config-manager --set-enabled powertools
+RUN dnf install -y epel-release
+RUN dnf check-update || true
+RUN dnf upgrade -y
+RUN dnf install -y python3 python3-pip python3-devel python3-jsonschema python3-pyyaml gdb vim git gcc gcc-c++ make cmake libuuid-devel json-c-devel hwloc-devel tbb-devel cli11-devel spdlog-devel libedit-devel systemd-devel doxygen python3-sphinx pandoc rpm-build rpmdevtools python3-virtualenv yaml-cpp-devel libudev-devel libcap-devel sudo
+
+RUN python3 -m pip install --user jsonschema virtualenv pudb pyyaml pybind11 && \
+    /usr/bin/python3 -m pip install setuptools --upgrade --prefix /usr && \
+    /usr/bin/python3 -m pip install --upgrade pip --prefix=/usr
```

Comment:
These are needed in order to avoid conflicts with our pyproject.toml implementation.  setuptools < ~51.x or so will fail to process the pyproject.toml successfully.  /usr is required so that the right Python installation is affected.  It has been noted to fail when only affecting the version in /usr/local.

Reply:
To add to what Tim said, the pip upgrade is needed because the build fails with pip < 10.0.0b1 (missing the command line option --no-warn-script-location), and the version installed through dnf is 9.x. I'll add a comment explaining both.